DeepMRSeg/deepmrseg_test.py

```diff
@@ -432,140 +461,387 @@ def _main():
+
+		# Create csv data
+		data=_np.array(['Scan1'] + FLAGS.inImg + [FLAGS.outImg]).reshape(-1, numMod+2)
+
+		# Write csv file
+		outCsv = _os.path.join(tmpDir, 'deepmrseg_test_filelist.csv')
+		with open(outCsv, 'w') as csvfile:
+			csvw = _csv.writer(csvfile)
+			csvw.writerows(columns)
+			csvw.writerows(data)
+
+		# Set flag for image list
+		FLAGS.sList = outCsv
+
+	if IOType == 2:
+
+		# Generate list of columns
+		numMod = 1          ## Multi-modal data is not implemented for "input from folder"
+		modCols = 'Mod_'
+		modCols = [modCols + str(i+1) for i in range(numMod)]
+		columns = _np.array(['ID'] + modCols + ['OutImg']).reshape(1,-1)
+
+		# Add absolute path to in/out dirs
+		FLAGS.inDir = _os.path.abspath(FLAGS.inDir)
+		FLAGS.outDir = _os.path.abspath(FLAGS.outDir)
+		
+		# Add image extension to out suffix (.nii.gz for now)
+		if FLAGS.outSuff.endswith('.nii.gz') == False:
+			FLAGS.outSuff = FLAGS.outSuff + '.nii.gz'
+			
+		# Create data frame
+		#   Get list of all .nii.gz images in the input dir
+		data = []
+		subInd = 1
+		for (dir, _, files) in _os.walk(FLAGS.inDir):
+			for i,f in enumerate(files):
+				if f.endswith('.nii.gz'):
+					pathIn = _os.path.join(dir, f)
+					pathOut = _os.path.join(FLAGS.outDir, f).replace('.nii.gz', FLAGS.outSuff)
+					data.append(['Scan' + str(subInd), pathIn, pathOut])
+					subInd += 1
+			
+		# Write csv file
+		outCsv = _os.path.join(tmpDir, 'deepmrseg_test_filelist.csv')
+		with open(outCsv, 'w') as csvfile:
+			csvw = _csv.writer(csvfile)
+			csvw.writerows(columns)
+			csvw.writerows(data)
+
+		# Set flag for image list
+		FLAGS.sList = outCsv
+	###############################################################
+
+
+	###############################################################
+	#### Run testing independently for each model path
+
+	num_models = len(FLAGS.mdlDir)
+	FLAGS.mdlDir = list(map(_os.path.abspath, FLAGS.mdlDir))   
+
+	#FOR
+	for indMdl, currMdl in enumerate(FLAGS.mdlDir):
 	
-	##########################
-	#### LOAD ALL MODELS #####
-	##########################
-	print("\n")
-	print("\n---->	Loading all stored models in ", FLAGS.mdlDir)
-	_sys.stdout.flush()
 	
-	allmodels = []
-	# Launch threads to load models simultaneously
-	print("")
-	#WITH
-	with _TPE( max_workers=None ) as executor:
-		# FOR ALL MODEL DIRS
-		for mDir in FLAGS.mdlDir:
+		##########################
+		#### LOAD ALL MODELS #####
+		##########################
+
+		## Load test_config.json to set orientation
+		testConf = (_os.path.join(currMdl, 'configs', 'test_config.json'))
+		roiIndFile = (_os.path.join(currMdl, 'configs', 'ROI_Indices.csv'))
+		with open(testConf) as f:
+			testconfigflags = _json.load(f)
+		trainflag_orient = testconfigflags['reorient']
+		trainflag_rescale = testconfigflags['rescale']
+		trainflag_ressize = testconfigflags['ressize']
+		trainflag_xy_width = testconfigflags['xy_width']
+		trainflag_num_classes = testconfigflags['num_classes']
+		
+
+		print("\n")
```

Comment:
Let's add an indent here to make the stdout prettier.

	      
		print("\n---->	Model " + str(indMdl +1))
		print("\t---->	Loading all stored models in model path " + str(indMdl+1) + ' : ' + currMdl)
		print("\t---->	Reorient is : " + trainflag_orient)

Reply:
Jimit, let's make a pass at the end to review/edit all print messages (including help messages, etc)